prow/plank/reconciler.go

```diff
@@ -233,6 +257,16 @@ func (r *reconciler) syncClusterStatus(interval time.Duration, knownClusters map
 							r.log.WithField("cluster", cluster).WithError(err).Warn("Error listing pod to check for build cluster reachability.")
 							status = ClusterStatusUnreachable
 						}
+
+						// Additionally check for pod verbs.
```

Comment:
Lets do the authorization check instead of listing pods rather than in addition to listing pods. The authorization check should cover the "List" verb on pods and it will avoid actually listing the pods which can be very expensive.

Reply:
I think the reason I did it this was was so that I didn't have to change any of the existing semantics around whether a cluster was reachable or not. If I get rid of the old listing pods behavior, then I think I should probably delete the `ClusterStatusUnreachable` status and replace it with `ClusterStatusMissingPermissions` or similar. IOW, I'm not sure how to best capture the "reachability" check when we only do the authz check. Thoughts?

Reply:
Listing pods was really just a means of checking the permissions so that sounds totally fine to me. I don't think we need to worry about changing the format of the cluster status file as long as `checkconfig` is updated accordingly (which is probably something this PR should include as well now that I think of it.).

One bit of naming feedback I'd add though is that "ErroringPermissions" might be better as "ErrorCheckingPermissions" or just "Error" since in that case we don't actually have a permissions problem.

Reply:
Here is the relevant part of `checkconfig`. https://github.com/kubernetes/test-infra/blob/a7466eca565b79cfce5d9bb7415f4fe5f8644383/prow/cmd/checkconfig/main.go#L1206-L1212  We should probably make this permissive of old and new values for a little while.

Reply:
I'm not sure if we actually need to update checkconfig at all. It currently only checks whether the status of the build cluster in question isn't `ClusterStatusReachable`. This is still the case after the changes in this PR so I think we're good? For example the tests in https://github.com/kubernetes/test-infra/blob/a7466eca565b79cfce5d9bb7415f4fe5f8644383/prow/cmd/checkconfig/main_test.go#L1857 reference the (no longer used) `ClusterStatusUnreachable` value but these tests still pass.